homeassistant/components/bmw_connected_drive/__init__.py

```diff
@@ -52,7 +53,12 @@
 
 CONFIG_SCHEMA = vol.Schema({DOMAIN: {cv.string: ACCOUNT_SCHEMA}}, extra=vol.ALLOW_EXTRA)
 
-SERVICE_SCHEMA = vol.Schema({vol.Required(ATTR_VIN): cv.string})
+SERVICE_SCHEMA = vol.Schema(
+    vol.Any(
+        {vol.Required(ATTR_VIN): cv.string},
+        {vol.Required(CONF_DEVICE_ID): cv.string},
+    )
```

Comment:
Use exclusive options in a group instead:

```suggestion
    vol.Exclusive(ATTR_VIN, "device"): cv.string,
    vol.Exclusive(CONF_DEVICE_ID, "device"): cv.string,
```

ideally, "device" is replaced by some kind of constant instead.

Reply:
`Exclusive` unfortunately allows no data to be sent, i.e. the `data` dict to be empty (see https://github.com/alecthomas/voluptuous/issues/115).
Of course I could handle a  `None` value in my code (which currently crashed if no data is available), but that would defeat the validation, wouldn't it?

Reply:
Hey, just checking if anything is missing or needs to be done for this PR to continue (except for waiting of course 🙂)